core/src/main/scala/akka/kafka/internal/MessageBuilder.scala

```diff
@@ -138,8 +138,9 @@ private[kafka] trait OffsetContextBuilder[K, V]
 )(
     val committer: KafkaAsyncConsumerCommitterRef
 ) extends CommittableOffsetMetadata {
-  override def commitScaladsl(): Future[Done] = committer.commitSingle(this)
-  override def commitJavadsl(): CompletionStage[Done] = commitScaladsl().toJava
+  override def commitScaladsl(): Future[Done] = commitInternal()
+  override def commitJavadsl(): CompletionStage[Done] = commitInternal().toJava
+  override def commitInternal(): Future[Done] = committer.commitSingle(this)
```

Comment:
Maybe just call it `commit` and mark/doc it as internal use.

Reply:
As these classes are marked as internal, I did not repeat the annotations.
The name is more of an extra safeguard, keep in mind that `private[kafka]` translates to `public` for Java.